widget/embedded/src/store/ui.ts

```diff
@@ -8,14 +12,45 @@ interface UiState {
   currentPage: string;
   setSelectedSwap: (requestId: string | null) => void;
   setCurrentPage: (path: string) => void;
+  isActiveTab: boolean | null;
+  showActivateTabModal: boolean;
+  activateCurrentTab: (
+    setCurrentTabAsActive: () => void,
+    hasRunningSwaps: boolean
+  ) => void;
+  setShowActivateTabModal: (flag: boolean) => void;
 }
-
 export const useUiStore = createSelectors(
-  create<UiState>()((set) => ({
+  create<UiState>()((set, get) => ({
     connectWalletsButtonDisabled: false,
     selectedSwapRequestId: null,
     currentPage: '',
+    isActiveTab: null,
+    showActivateTabModal: false,
     setSelectedSwap: (requestId) => set({ selectedSwapRequestId: requestId }),
     setCurrentPage: (path) => set({ currentPage: path }),
+    activateCurrentTab: (setCurrentTabAsActive, hasRunningSwaps) => {
+      const { showActivateTabModal } = get();
+
+      if (!showActivateTabModal && hasRunningSwaps) {
+        set({ showActivateTabModal: true });
+      } else if (showActivateTabModal || !hasRunningSwaps) {
+        setCurrentTabAsActive();
+      }
+    },
+    setShowActivateTabModal: (flag) => {
+      set({ showActivateTabModal: flag });
+    },
   }))
 );
+
+const tabManager: TabManagerInterface = new TabManager({
```

Comment:
Please move this piece of code to a custom hook called `useTabManager`, because it's using zustand's state which I guess should be run in a react component. If it is, please to avoid misusage in future by calling it outside React, Move it to a hook.